src/probability/independence.lean

```diff
@@ -463,41 +469,53 @@ section indep_fun
 variables {α β β' γ γ' : Type*} {mα : measurable_space α} {μ : measure α} {f : α → β} {g : α → β'}
 
 lemma indep_fun_iff_measure_inter_preimage_eq_mul
-  {mβ : measurable_space β} {mβ' : measurable_space β'} [is_probability_measure μ]
-  (hf : measurable f) (hg : measurable g) :
+  {mβ : measurable_space β} {mβ' : measurable_space β'} :
   indep_fun f g μ
     ↔ ∀ s t, measurable_set s → measurable_set t
       → μ (f ⁻¹' s ∩ g ⁻¹' t) = μ (f ⁻¹' s) * μ (g ⁻¹' t) :=
 begin
-  let Sf := {t1 | ∃ s, measurable_set s ∧ f ⁻¹' s = t1},
-  let Sg := {t1 | ∃ t, measurable_set t ∧ g ⁻¹' t = t1},
-  suffices : indep_fun f g μ ↔ indep_sets Sf Sg μ,
-  { refine this.trans _,
-    simp_rw [indep_sets, Sf, Sg, set.mem_set_of_eq],
-    split; intro h,
-    { exact λ s t hs ht, h (f ⁻¹' s) (g ⁻¹' t) ⟨s, hs, rfl⟩ ⟨t, ht, rfl⟩, },
-    { rintros t1 t2 ⟨s, hs, rfl⟩ ⟨t, ht, rfl⟩,
-      exact h s t hs ht, }, },
-  have hSf_pi : is_pi_system Sf,
-  { rintros s ⟨s', hs', rfl⟩ t ⟨t', ht', rfl⟩ hst_nonempty,
-    exact ⟨s' ∩ t', hs'.inter ht', rfl⟩, },
-  have hSg_pi : is_pi_system Sg,
-  { rintros s ⟨s', hs', rfl⟩ t ⟨t', ht', rfl⟩ hst_nonempty,
-    exact ⟨s' ∩ t', hs'.inter ht', rfl⟩, },
-  have hSf_gen : mβ.comap f = generate_from Sf := mβ.comap_eq_generate_from f,
-  have hSg_gen : mβ'.comap g = generate_from Sg := mβ'.comap_eq_generate_from g,
-  rw indep_fun,
   split; intro h,
-  { rw [hSf_gen, hSg_gen] at h,
-    exact indep.indep_sets h, },
-  { exact indep_sets.indep hf.comap_le hg.comap_le hSf_pi hSg_pi hSf_gen hSg_gen h, },
+  { refine λ s t hs ht, h (f ⁻¹' s) (g ⁻¹' t) ⟨s, hs, rfl⟩ ⟨t, ht, rfl⟩, },
+  { rintros _ _ ⟨s, hs, rfl⟩ ⟨t, ht, rfl⟩, exact h s t hs ht, },
+end
+
+lemma Indep_fun_iff_measure_inter_preimage_eq_mul {ι : Type*} {β : ι → Type*}
+  (m : Π x, measurable_space (β x)) (f : Π i, α → β i) :
+  Indep_fun m f μ
+    ↔ ∀ (S : finset ι) {sets : Π i : ι, set (β i)} (H : ∀ i, i ∈ S → measurable_set[m i] (sets i)),
+      μ (⋂ i ∈ S, (f i) ⁻¹' (sets i)) = ∏ i in S, μ ((f i) ⁻¹' (sets i)) :=
+begin
+  split; intro h,
+  { refine λ S sets h_meas, h _ (λ i hi_mem, ⟨sets i, h_meas i hi_mem, rfl⟩), },
```

Comment:
```suggestion
  refine ⟨λ h S sets h_meas, h _ (λ i hi_mem, ⟨sets i, h_meas i hi_mem, rfl⟩), λ h, _⟩,
```
It's exactly the same proof, but then you can remove one indentation level from the rest of the proof.